### DISCUSSION TITLE
quasi neutrality with constant momentum distribution

### DISCUSSION THREAD

**Discussion by floresv299** [BLAST-WarpX/warpx#4445] [Simulation Inputs] [answered]
Hello all, 

I'm trying to replicate some findings that I got using OSIRIS. My simulation is simple: it is a 1D3V simulation involving a laser interacting with a solid target (10s to 100s of times the critical density). My density profile has an exponential scale length to model the pre-plasma. The 1D3V aspect of my simulation comes from a constant momentum drift in the -y direction (p =-$\hat{y}$ mc tan$\theta$). This sets up a "boosted frame", essentially allowing interactions to be oblique (with other spatial and temporal transformations). Currently, when I apply the momentum drift I get different densities for the ions and electrons. I was thinking that since both species have the same constant momentum then quasi-neutrality will be maintained like in my previous simulations. However, this seems not to be the case. Any advice ? Below I have attached my inputs and a plot of the densities. Thank you all ! 
[Warpx_Momentum_Drift.pdf](https://github.com/ECP-WarpX/WarpX/files/13469062/Warpx_Momentum_Drift.pdf)
[Warpx_mom_drift.txt](https://github.com/ECP-WarpX/WarpX/files/13469074/Warpx_mom_drift.txt)

**Comment by RemiLehe** [comment]
Hi, 
Sometimes, in PIC simulation, a neutral plasma can become **numerically** unstable when it is drifting (even though it should be **physically** stable). The numerical stability usually depends on many parameters, including the density of the plasma, the drift velocity, the grid resolution and the type of solver (e.g. Yee vs other solvers).
I am not too familiar with the stability properties of the the 1D case with a drift in `y`, but do you have a way to check whether this should in principle be numerically stable. For instance, have you run the exact same simulation (i.e. same density, same resolution) with Osiris?

**Reply by floresv299** [reply, marked as answer]
Hi Remi, 

Thank you so much for the help. Turns out it was a resolution problem. Increasing the resolution did the job ! Thanks for your insight. 

best,

Victor
